### PR TITLE
Feature: Quiz a specific question.

### PR DESCRIPTION
**Pull request by aryagiat**
Users can pick a question from the list and quiz themselves based on the question. 

Command usage: `quiz [INDEX]`

### CODE REVIEW

File: src/main/java/seedu/smartnus/model/question/predicate/ShowQuestionIndexPredicate.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    /**
+     * Constructor for the predicate.
+     * @param targetIndex The target index of the list.
+     */
+    public ShowQuestionIndexPredicate(Index targetIndex) {
+        this.currentIndex = 0;
+        this.targetIndex = targetIndex;
+    }
+
```
Comment (Agentum07) [inline src/main/java/seedu/smartnus/model/question/predicate/ShowQuestionIndexPredicate.java:24]: Hi is it possible to write a test to cover this part?

File: src/main/java/seedu/smartnus/model/question/predicate/ShowQuestionIndexPredicate.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    @Override
+    public boolean test(Question question) {
+        if (currentIndex == targetIndex.getZeroBased()) {
+            currentIndex++;
+            return true;
+        } else {
```
Comment (Agentum07) [inline src/main/java/seedu/smartnus/model/question/predicate/ShowQuestionIndexPredicate.java:30]: Can you write a test to cover this?

Comment (aryagiat) [reply src/main/java/seedu/smartnus/model/question/predicate/ShowQuestionIndexPredicate.java:30]: done

File: src/main/java/seedu/smartnus/model/question/predicate/ShowQuestionIndexPredicate.java (hunk 1 above)
Comment (aryagiat) [reply src/main/java/seedu/smartnus/model/question/predicate/ShowQuestionIndexPredicate.java:24]: done

**Pull request merged**